mypy/constant_fold.py

```diff
@@ -100,6 +148,36 @@ def constant_fold_binary_int_op(op: str, left: int, right: int) -> int | None:
     return None
 
 
+def constant_fold_binary_float_op(op: str, left: int | float, right: int | float) -> float | None:
+    assert not (isinstance(left, int) and isinstance(right, int)), (op, left, right)
+    if op == "+":
+        return left + right
+    elif op == "-":
+        return left - right
+    elif op == "*":
+        return left * right
+    elif op == "/":
+        if right != 0:
+            return left / right
+    elif op == "//":
+        if right != 0:
+            return left // right
+    elif op == "%":
+        if right != 0:
+            return left % right
+    elif op == "**":
+        if (left < 0 and right >= 1 or right == 0) or (left >= 0 and right >= 0):
+            try:
+                ret = left**right
+            except OverflowError:
+                return None
```

Comment:
Digging through the CPython source, [only power ops can raise an OverflowError](https://github.com/python/cpython/blob/7894bbe94ba319eb650f383cb5196424c77b2cfd/Objects/floatobject.c#L852).